source/jormungandr/jormungandr/parking_space_availability/bss/cykleo.py

```diff
@@ -0,0 +1,147 @@
+
+# Copyright (c) 2001-2014, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from __future__ import absolute_import, print_function, unicode_literals, division
+from jormungandr.parking_space_availability import AbstractParkingPlacesProvider
+from jormungandr import cache, app
+import pybreaker
+import logging
+import json
+import requests as requests
+from jormungandr.ptref import FeedPublisher
+from jormungandr.parking_space_availability.bss.stands import Stands
+
+
+DEFAULT_CYKLEO_FEED_PUBLISHER = {
+    'id': 'cykleo',
+    'name': 'cykleo',
+    'license': 'Private',
+    'url': 'www.cykleo.fr'
+}
+
+
+class CykleoProvider(AbstractParkingPlacesProvider):
+    def __init__(self, url, network, username, password, operators={'cykleo'}, certifi_verify=False,
+                 service_id=None, timeout=2,
+                 feed_publisher=DEFAULT_CYKLEO_FEED_PUBLISHER, **kwargs):
+        self.url = url
+        self.network = network.lower()
+        self.service_id = service_id
+        self.certifi_verify = certifi_verify
+        self.username = username
+        self.password = password
+        self.operators = [o.lower() for o in operators]
+        self.timeout = timeout
+        self.breaker = pybreaker.CircuitBreaker(
+            fail_max=kwargs.get('circuit_breaker_max_fail', app.config['CIRCUIT_BREAKER_MAX_CYKLEO_FAIL']),
+            reset_timeout=kwargs.get('circuit_breaker_reset_timeout', app.config['CIRCUIT_BREAKER_CYKLEO_TIMEOUT_S']))
+        self._feed_publisher = FeedPublisher(**feed_publisher) if feed_publisher else None
+
+    def service_caller(self, method, url, data=None, headers=None):
+        try:
+            kwargs = {"timeout": self.timeout, "verify": self.certifi_verify}
+            if headers:
+                kwargs.update({"headers": headers})
+            if data:
```

Comment:
Not sure the case can happen but I'd rather have `if data is not None` (same above for headers).

Reply:
In that case: if data is not None == if data